src/datanode/src/sql/insert.rs

```diff
@@ -11,28 +11,48 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+use std::collections::HashMap;
+use std::pin::Pin;
 
 use catalog::CatalogManagerRef;
 use common_query::Output;
+use common_recordbatch::{RecordBatch, RecordBatches};
+use datafusion_expr::type_coercion::binary::coerce_types;
+use datafusion_expr::Operator;
 use datatypes::data_type::DataType;
 use datatypes::schema::ColumnSchema;
 use datatypes::vectors::MutableVector;
+use futures::stream::{self, StreamExt};
+use futures::Stream;
+use query::parser::QueryStatement;
+use session::context::QueryContextRef;
 use snafu::{ensure, OptionExt, ResultExt};
 use sql::ast::Value as SqlValue;
 use sql::statements::insert::Insert;
+use sql::statements::statement::Statement;
 use sql::statements::{self};
 use table::engine::TableReference;
 use table::requests::*;
+use table::TableRef;
 
 use crate::error::{
-    CatalogSnafu, ColumnDefaultValueSnafu, ColumnNoneDefaultValueSnafu, ColumnNotFoundSnafu,
-    ColumnValuesNumberMismatchSnafu, InsertSnafu, ParseSqlSnafu, ParseSqlValueSnafu, Result,
-    TableNotFoundSnafu,
+    CatalogSnafu, CollectRecordsSnafu, ColumnDefaultValueSnafu, ColumnNoneDefaultValueSnafu,
+    ColumnNotFoundSnafu, ColumnTypeMismatchSnafu, ColumnValuesNumberMismatchSnafu, Error,
+    ExecuteSqlSnafu, InsertSnafu, MissingInsertBodySnafu, ParseSqlSnafu, ParseSqlValueSnafu,
+    Result, TableNotFoundSnafu,
 };
-use crate::sql::{SqlHandler, SqlRequest};
+use crate::sql::{table_idents_to_full_name, SqlHandler, SqlRequest};
 
 const DEFAULT_PLACEHOLDER_VALUE: &str = "default";
 
+type InsertRequstStream = Pin<Box<dyn Stream<Item = Result<SqlRequest>> + Send>>;
+pub(crate) enum InsertRequests {
+    // Single request
+    Request(SqlRequest),
+    // Streaming requests
+    Stream(InsertRequstStream),
```

Comment:
typo

Reply:
We are good here. the query part of the insert stmt will goes into the standard procedure of parsing a Query, which would invoke `get_table_provider` as a normal Query would do

Add something like `println!("get_table_provider: {:?}", name);` in `get_table_provider` and it should be easy to try in standalone mode